File: 5kyu/(5 kyu) RGB To Hex Conversion/(5 kyu) RGB To Hex Conversion.cpp
#include <sstream>
#include <iomanip>

class RGBToHex
{
  static inline std::string dec_to_hex(int c)
  {
    if (c < 0) return "00";
    else if (c > 255) return "FF";
    std::ostringstream oss;
    oss << std::uppercase << std::setfill('0') << std::hex << std::setw(2) << c;
    return oss.str();
  }
  
  public:
  static std::string rgb(int r, int g, int b)
  {
    return dec_to_hex(r) + dec_to_hex(g) + dec_to_hex(b);
  }
};

File: 5kyu/(5 kyu) RGB To Hex Conversion/(5 kyu) RGB To Hex Conversion.rs
fn rgb(r: i32, g: i32, b: i32) -> String {
  fn dec_to_hex(c: i32) -> String {
    if c < 0 { String::from("00") }
    else if c > 255 { String::from("FF") }
    else { format!("{:02X}", c) }
  }

  format!("{}{}{}", dec_to_hex(r), dec_to_hex(g), dec_to_hex(b))
}
